Assets/Scripts/Enemy.cs

```diff
@@ -1,35 +1,114 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using System;
 
-public class Enemy : MonoBehaviour
-{
-
-    private Rigidbody2D myRigidbody;
-    private SpriteRenderer mySprite;
-    public Rigidbody2D player;
-    // Start is called before the first frame update
-    public float max_speed = 4;
-    public float acceleration_rate = 1;
-    private void Awake(){
-        myRigidbody = GetComponent<Rigidbody2D>();
-        mySprite = GetComponent<SpriteRenderer>();
-    }
-    void Start()
-    {
-        player = GameObject.FindGameObjectWithTag("Player").GetComponent<Rigidbody2D>();
+/// <summary>
+/// Abstract class providing framework for Enemy design
+/// </summary>
+public abstract class Enemy : MonoBehaviour {
+    /// <summary>
+    /// Dictionary of EnemyBehaviours and animation functions to be used by Enemy in associated ENEMY_STATE
+    /// </summary>
+    protected IDictionary<ENEMY_STATE, (EnemyBehaviour, Action)> _behaviours = new Dictionary<ENEMY_STATE, (EnemyBehaviour, Action)>();
```

Comment:
I've now had to change the dictionary to one that stores not just EnemyBehaviours, but tuples of EnemyBehaviours and Actions. This is because now we need to take an ENEMY_STATE and perform **not just** a corresponding EnemyBehaviour **but also** an animation function.

I had two obvious choices in how I could implement this. I could select the animation function using a switch on ENEMY_STATE values or I could use a dictionary lookup. I defaulted to a dictionary lookup for readability, and in order to perform only one dictionary lookup instead of two, I expanded the values of our preexisting dictionary instead of making a second new dictionary.

Reply:
I forgot to @Micky774 even though you're the one whose eyeballs I need on this comment and the corresponding comment below.

Reply:
This is no longer the current implementation, and as such is outdated and resolved.